DiscordChatExporter.Core/Exporting/ExportAssetDownloader.cs

```diff
@@ -51,24 +52,31 @@ await Http.ResiliencePipeline.ExecuteAsync(
                 // Try to set the file date according to the last-modified header
                 try
                 {
-                    var lastModified = response
-                        .Content.Headers.TryGetValue("Last-Modified")
-                        ?.Pipe(s =>
-                            DateTimeOffset.TryParse(
-                                s,
-                                CultureInfo.InvariantCulture,
-                                DateTimeStyles.None,
-                                out var instant
-                            )
-                                ? instant
-                                : (DateTimeOffset?)null
-                        );
-
-                    if (lastModified is not null)
+                    if (timestamp is null)
                     {
-                        File.SetCreationTimeUtc(filePath, lastModified.Value.UtcDateTime);
-                        File.SetLastWriteTimeUtc(filePath, lastModified.Value.UtcDateTime);
-                        File.SetLastAccessTimeUtc(filePath, lastModified.Value.UtcDateTime);
+                        var lastModified = response
+                            .Content.Headers.TryGetValue("Last-Modified")
+                            ?.Pipe(s =>
+                                DateTimeOffset.TryParse(
+                                    s,
+                                    CultureInfo.InvariantCulture,
+                                    DateTimeStyles.None,
+                                    out var instant
+                                )
+                                    ? instant
+                                    : (DateTimeOffset?)null
+                            );
+                        if (lastModified is not null)
+                        {
+                            timestamp = lastModified;
+                        }
```

Comment:
If the header option is no longer reliable, I'd prefer we remove this codepath entirely to simplify things.